runtime/amd64.S

```diff
@@ -65,6 +65,13 @@
 
 #endif
 
+#if defined(SYS_linux) || defined(SYS_gnu)
```

Comment:
Is this `#if` necessary?

Reply:
It matches the check before .size is emitted for regular functions: https://github.com/ocaml/ocaml/blob/6f9eb6ff0e0b553f6229748abfd4a2f366676be8/asmcomp/amd64/emit.mlp#L938-L940

Similarly, for i386 backend:
https://github.com/ocaml/ocaml/blob/6f9eb6ff0e0b553f6229748abfd4a2f366676be8/asmcomp/i386/emit.mlp#L907-L909

I don't know why other systems don't use .size.

Reply:
OK, fair enough. I think at least `mingw/mingw64` should also accept this directive, but this is unrelated to this PR, so can be left for later.

Reply:
Thank you!